iceoryx_binding_c/include/iceoryx_binding_c/types.h

```diff
@@ -18,8 +18,32 @@
 #ifndef IOX_BINDING_C_TYPES_H
 #define IOX_BINDING_C_TYPES_H
 
+#include "iceoryx_binding_c/iceoryx_binding_c_deployment.h"
 #include "internal/c2cpp_binding.h"
 
+/// @brief if the function parameters change due to an update of the listener or
+///        waitset recalculate them with the following approach.
+///        1. Run SanityCheck.SanityCheck.VerifyStorageSizeCalculationForListener
+///            or SanityCheck.VerifyStorageSizeCalculationForWaitSet
+///        2. Take a look at the expected numbers of size 1 (A1) and 2 (A2).
+///        3. Find the parameters m, n for the function StorageSize(x) = m + n * x
+///        4. Re-run the the tests to verify if the parameters are correct.
+#if defined(__APPLE__)
+#define CALCULATE_STORAGE_SIZE_FOR_LISTENER(numberOfAttachments)                                                       \
```

Comment:
I do not believe any of these formulas will do us any good since they assume too much about how the compiler will structure the data. It is true that there must be a lower bound, but theoretically there does not have to be an upper bound as the compiler is allowed to leave empty bytes (alignment) etc. for whatever reason. It will of course try to conserve space and minimize these empty bytes, but we just cannot know the exact number.

It is hard to get a (close) upper bound, and in general not possible without compiler specific knowledge.